parsing/ast_iterator.ml

```diff
@@ -259,7 +277,8 @@ module MT = struct
     | Psig_class_type l ->
         List.iter (sub.class_type_declaration sub) l
     | Psig_extension (x, attrs) ->
-        sub.extension sub x; sub.attributes sub attrs
+        sub.attributes sub attrs;
+        sub.extension sub x
```

Comment:
I'm not sure if there are consequences to this order change, but it is more consistent with other cases after this change.

Reply:
The idea is to have `.attributes` come right after `.location` for all AST nodes, so that you can know when you process an attribute the location of its whole attached node (attribute included) by an ugly side-effect. See #1765. This is fragile of course, and would like a better long-term way to do it.